src/lib/kdf/prf_tls/prf_tls.cpp

```diff
@@ -58,21 +55,12 @@ std::unique_ptr<KDF> TLS_12_PRF::new_object() const {
    return std::make_unique<TLS_12_PRF>(m_mac->new_object());
 }
 
-void TLS_12_PRF::kdf(uint8_t key[],
-                     size_t key_len,
-                     const uint8_t secret[],
-                     size_t secret_len,
-                     const uint8_t salt[],
-                     size_t salt_len,
-                     const uint8_t label[],
-                     size_t label_len) const {
-   secure_vector<uint8_t> msg;
-
-   msg.reserve(label_len + salt_len);
-   msg += std::make_pair(label, label_len);
-   msg += std::make_pair(salt, salt_len);
-
-   P_hash(key, key_len, *m_mac, secret, secret_len, msg.data(), msg.size());
+void TLS_12_PRF::perform_kdf(std::span<uint8_t> key,
+                             std::span<const uint8_t> secret,
+                             std::span<const uint8_t> salt,
+                             std::span<const uint8_t> label) const {
+   const auto msg = concat<secure_vector<uint8_t>>(label, salt);
```

Comment:
No real reason for `secure_vector` here, label is a constant and the salts are in the public hello messages

Reply:
Alternately could amend `P_hash` to take 2 arguments for these values

Reply:
I removed `P_hash` entirely and moved its implementation straight into `perform_kdf`.

Reply:
> I removed P_hash entirely 

Oh right that function only existed so that the TLS 1.0 and TLS 1.2 KDFs could share code, no need for it anymore now that TLS 1.0 is dead.